### PR TITLE
[action] expose chainID in the transaction

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
1. in v1.8, we add a hard-fork protection to make sure all nodes accept/expose chainID at the same height
2. this protection can be removed post v1.8

## Type of change
- [x] Code refactor or improvement

# How Has This Been Tested?
- [x] make test
- [x] fullsync

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: action/action_deserializer.go [hunk 1]
```diff
@@ -6,31 +6,15 @@
 
 package action
 
-import (
-	"github.com/iotexproject/iotex-proto/golang/iotextypes"
-)
+import "github.com/iotexproject/iotex-proto/golang/iotextypes"
 
 // Deserializer de-serializes an action
 type Deserializer struct {
```
Comment (Liuhaai) [inline action/action_deserializer.go:12]: Can this struct be removed?

**Comment by Liuhaai** [comment]
other seems good for me

File: action/action_deserializer.go (hunk 1 above)
Comment (dustinxie) [reply action/action_deserializer.go:12]: this struct will be used in later PR to de-couple the global `evmNetworkID`, so we keep it

**Comment by dustinxie** [comment]
> how do we do 'fullsync' test before v1.8.0 activation?

1. the test is done during the weekend, to sync to the latest tip
2. then I have a GCP mainnet node, running this image, it successfully passed the 180 height

File: action/action_deserializer.go [hunk 2]
```diff
@@ -6,31 +6,15 @@
 
 package action
 
-import (
-	"github.com/iotexproject/iotex-proto/golang/iotextypes"
-)
+import "github.com/iotexproject/iotex-proto/golang/iotextypes"
 
 // Deserializer de-serializes an action
```
Comment (Liuhaai) [inline action/action_deserializer.go:11]: Pls add some explanatory comments for `action_deserializer` and `block_deserializer`

**Review by Liuhaai — APPROVED** [review]

**Comment by CoderZhi** [comment]
> > how do we do 'fullsync' test before v1.8.0 activation?
> 
> 1. the test is done during the weekend, to sync to the latest tip (at then-time, before 180 height)
> 2. then I have a GCP mainnet node, running this image, it successfully passed the 180 height

good to know

**Pull request merged**